frontend/src/employee-frontend/components/vasu/VasuPage.tsx

```diff
@@ -0,0 +1,234 @@
+                question1.value = checked
+                return clone
+              })
+            })
+          }}
+        />
+      </>
+    )
+  }
+  function renderRadioGroupQuestion(
+    question: RadioGroupQuestion,
+    sectionIndex: number,
+    questionIndex: number
+  ) {
+    return (
+      <>
+        <Label>{question.name}</Label>
+        <Gap size={'m'} />
+        <FixedSpaceColumn>
+          {vasuDocument.isSuccess &&
+            question.optionNames.map((optionName, optionIndex) => {
+              const question1 = vasuDocument.value.content.sections[
+                sectionIndex
+              ].questions[questionIndex] as RadioGroupQuestion
+              return (
+                <Radio
+                  key={optionName}
+                  checked={question1.value === optionIndex}
+                  label={optionName}
+                  onChange={() => {
+                    setVasuDocument((prev) => {
+                      return prev.map((prevData) => {
+                        const clone = cloneDeep(prevData)
+                        const question1 = clone.content.sections[sectionIndex]
+                          .questions[questionIndex] as RadioGroupQuestion
+                        question1.value = optionIndex
+                        return clone
+                      })
+                    })
+                  }}
+                />
+              )
+            })}
+        </FixedSpaceColumn>
+      </>
+    )
+  }
+  function renderMultiSelectQuestion(
+    question: MultiSelectQuestion,
+    sectionIndex: number,
+    questionIndex: number
+  ) {
+    return (
+      <>
+        <Label>{question.name}</Label>
+        <Gap size={'m'} />
+        <FixedSpaceColumn>
+          {vasuDocument.isSuccess &&
+            question.optionNames.map((optionName, optionIndex) => {
+              const question1 = vasuDocument.value.content.sections[
+                sectionIndex
+              ].questions[questionIndex] as MultiSelectQuestion
+              return (
+                <Checkbox
+                  key={optionName}
+                  checked={question1.value.includes(optionIndex)}
+                  label={optionName}
+                  onChange={() => {
+                    setVasuDocument((prev) => {
+                      return prev.map((prevData) => {
+                        const clone = cloneDeep(prevData)
+                        const question1 = clone.content.sections[sectionIndex]
+                          .questions[questionIndex] as MultiSelectQuestion
+                        if (question1.value.includes(optionIndex)) {
+                          question1.value = question.value.filter(
+                            (elem) => elem !== optionIndex
+                          )
+                        } else {
+                          question1.value.push(optionIndex)
+                        }
+                        return clone
```

Comment:
this probably doesn't matter, but onChange callback function has parameter `checked` and using it could be more reliable than just toggling based on current value
```
if(checked && !question1.value.includes(optionIndex)) question1.value.push(optionIndex)
if(!checked) question1.value = question1.value.filter(i => i !== optionIndex)
```
both ways likely work though